bianca_pipes.rb

```diff
@@ -0,0 +1,202 @@
+##### Wave_1 #####
+
+#create SolarSystem class
+
+class SolarSystem
+
+  #use @planets instance variable
+  def initialize(planets)
+    @planets = Array.new(planets)
+  end
+
+  #method to add planet to list
+
+  def add(planet)
+    @planet = planet
+    @planets << planet
+    return @planet
+  end
+
+  #method to return, not print list of planets as String
+  def list
+    num = 0
+    planet_list = ""
+    @planets.each do |planet|
+      num += 1
+      planet_list += "#{num}. #{planet}\n"
+    end
+    ## Not sure why return does not work here instead of puts ##
+    puts planet_list
+  end
+end
+
+milky_way = SolarSystem.new(milky_way)
+
+milky_way.add("Mercury")
+milky_way.add("Venus")
+milky_way.add("Earth")
+
+milky_way.list
+
+milky_way.add("Hoth")
+
+milky_way.list
+
+class SolarSystem
+
+def initialize(planets, planet_name, mass, orbit, tilt, rotation, num_of_moons)
+  @planets = planets
+  @planets = Hash.new
+  @planet_name, @mass, @orbit, @tilt, @rotation, @num_of_moons  = planet_name, mass, orbit, tilt, rotation, num_of_moons
+  @planets[:name] = planet_name
+  @planets[:mass] = mass
+  @planets[:orbit] = orbit
+  @planets[:tilt] = tilt
+  @planets[:rotation] = rotation
+  @planets[:num_of_moons] = num_of_moons
+end
+
+def add_hash(planet_name, mass, orbit, tilt, rotation, num_of_moons)
+  @planets[:name] = planet_name
+  @planets[:mass] = mass
+  @planets[:orbit] = orbit
+  @planets[:tilt] = tilt
+  @planets[:rotation] = rotation
+  @planets[:num_of_moons] = num_of_moons
+end
+
+#return list of planets as string
+def return_planets
+  planet_hash = ""
+  @planets.each do |key, value|
+    planet_hash += "#{key}: #{value}\n"
+  end
+  puts planet_hash
+end
+end
+
+milky_way = SolarSystem.new("milkyway", "Earth", 34, 5, 6, 7, 1)
+
+milky_way.return_planets
+
+####Wave_2 & Wave_3 #####
+
+class SolarSystem
+
+  #use @planets instance variable
+  def initialize(planets)
+    @planets = Array.new(planets)
+  end
+
+  #method to add planet to list
+  def add(planet)
+    @planet = planet
+    @planets << planet
+    return @planet
+  end
+
+  #create interface for user to interact with solar system
+  #can select planet and view info about it
+
+  def interact
+    puts "What planet would you like to view and learn about?"
+    planet = gets.chomp
+    @planets.each do |series_of_planets|
+      if series_of_planets == planet
+        puts "#{planet.name}"
+        puts "#{planet.weight}"
+        puts "#{planet.size}"
+        puts "#{planet.orbit}"
+      else
+        puts "Sorry, it appears that the planet you want is not available."
+      end
+    end
+  end
+
+  #method to return, not print list of planets as String
+  def list
+    num = 0
+    planet_list = ""
```

Comment:
I really like that you've taken all the code to interact with the user and split it into separate methods. It makes your driver code below very concise and easy to read.